dashboard/src/components/Summary/Summary.tsx

```diff
@@ -67,28 +73,57 @@ const Summary = ({
   );
 };
 
+const LinkWrapper = ({
+  filterSection,
+  filterValue,
+}: {
+  filterSection: TFilterObjectsKeys;
+  filterValue: string;
+}): JSX.Element => {
+  const diffFilter = useDiffFilterParams(filterValue, filterSection);
+
+  return (
+    <Link
+      search={previousParams => ({
+        ...previousParams,
+        diffFilter,
+      })}
+      key={filterValue}
+      className="line-clamp-1"
+    >
+      {filterValue}
+    </Link>
+  );
+};
+
 export const SummaryItem = ({
   arch,
   compilers,
   onClickKey,
-  onClickCompiler,
   leftIcon,
 }: ISummaryItem): JSX.Element => {
+  const diffFilter = useDiffFilterParams(arch.text, 'archs');
+
   const compilersElement = useMemo(() => {
     return compilers?.map(compiler => (
-      <button
+      <LinkWrapper
         key={compiler}
-        className="line-clamp-1"
-        onClick={() => onClickCompiler?.(compiler)}
-      >
-        {compiler}
-      </button>
+        filterSection="compilers"
+        filterValue={compiler}
+      />
```

Comment:
The component's name is LinkWrapper but in this case, it doesn't wrap anything.

In my opinion, it would be more interesting to use the filterValue as a children, or change the component name